### DISCUSSION TITLE
Its not possible to read params in server components, WHYY?

### DISCUSSION THREAD

**Discussion by jiri-grussmann** [vercel/next.js#66862] [App Router]
It's convenient that the fetch request is cached during a single render, allowing multiple components to call it while only making one actual request. However, since most requests depend on URL parameters, I still need to pass props down to the components that need to make the call.

Consequently, the idea of calling the request in every component becomes somewhat pointless, as I end up drilling down all the necessary props anyway. Therefore, I might as well directly pass down the request results.

So is there any way to escape this hell? Because it seems like this is an absolutely crucial issue that hasn't been resolved at all.

**Comment by udayshankark-qp** [comment]
In server the main page server components you can get the params using `searchParams` prop.
Instead of passing the searchParams to child client components, you can always use the `useSearchParams` hook in only the components needed to fetch the params

**Reply by jiri-grussmann** [reply]
Its the server components I am talking about, I dont care about getting it on client I know I can use the hook there, I need to get it on a server, not only in client components.. (given the scenario I did describe in the post above..)

**Reply by George-Acquah** [reply]
sure, you can get the params using the searchParams props.
In your server component, you can do something like this
```
interface ISearchParams {
  searchParams?: {
    query?: string;
    page?: string;
    size?: string;
  };
}

export default async function ServerComponent({ searchParams }: ISearchParams) {
  const query= searchParams?.query || "";
  const currentPage = Number(searchParams?.page) || 1;
  const pageSize = Number(searchParams?.size) || 5;
  }
```

**Reply by jiri-grussmann** [reply]
Yes, but this only works in page component not a component in a subtree

**Reply by enzzzooo** [reply]
> Yes, but this only works in page component not a component in a subtree

how about passing the searchParams as props through the tree?
